File: schemaorggo/question.go
package schemaorggo

import "encoding/json"

// Question see : https://schema.org/Question
type Question struct {
	CreativeWork

	typeContext

	// AcceptedAnswer see : https://schema.org/acceptedAnswer
	// The answer that has been accepted as best, typically on a Question/Answer site. Sites vary in their selection mechanisms, e.g. drawing on community opinion and/or the view of the Question author.
	// types : Answer
	AcceptedAnswer []*Answer `json:"acceptedAnswer,omitempty"`

	// AnswerCount see : https://schema.org/answerCount
	// The number of answers this question has received.
	// types : Integer
	AnswerCount []float64 `json:"answerCount,omitempty"`

	// DownvoteCount see : https://schema.org/downvoteCount
	// The number of downvotes this question, answer or comment has received from the community.
	// types : Integer
	DownvoteCount []float64 `json:"downvoteCount,omitempty"`

	// SuggestedAnswer see : https://schema.org/suggestedAnswer
	// An answer (possibly one of several, possibly incorrect) to a Question, e.g. on a Question/Answer site.
	// types : Answer
	SuggestedAnswer []*Answer `json:"suggestedAnswer,omitempty"`

	// UpvoteCount see : https://schema.org/upvoteCount
	// The number of upvotes this question, answer or comment has received from the community.
	// types : Integer
	UpvoteCount []float64 `json:"upvoteCount,omitempty"`
}

func (v Question) intoMap(intop *map[string]interface{}) error {
	if intop == nil {
		return nil
	}

	v.CreativeWork.intoMap(intop)

	into := *intop

	{
		var value interface{} = v.AcceptedAnswer
		if len(v.AcceptedAnswer) == 1 {
			value = v.AcceptedAnswer[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["acceptedAnswer"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.AnswerCount
		if len(v.AnswerCount) == 1 {
			value = v.AnswerCount[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["answerCount"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.DownvoteCount
		if len(v.DownvoteCount) == 1 {
			value = v.DownvoteCount[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["downvoteCount"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.SuggestedAnswer
		if len(v.SuggestedAnswer) == 1 {
			value = v.SuggestedAnswer[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["suggestedAnswer"] = json.RawMessage(b)
		}
	}

	{
		var value interface{} = v.UpvoteCount
		if len(v.UpvoteCount) == 1 {
			value = v.UpvoteCount[0]
		}

		b, err := json.Marshal(value)
		if err != nil {
			return err
		}

		if len(b) > 0 && string(b) != "null" {
			into["upvoteCount"] = json.RawMessage(b)
		}
	}

	*intop = into

	return nil
}

func (v Question) asMap() (map[string]interface{}, error) {
	data := map[string]interface{}{}
	err := v.intoMap(&data)
	if err != nil {
		return nil, err
	}

	data["@context"] = "http://schema.org"
	data["@type"] = "Question"

	return data, nil
}

func (v Question) MarshalJSON() ([]byte, error) {
	data, err := v.asMap()
	if err != nil {
		return nil, err
	}

	return json.Marshal(data)
}

File: schemaorgphp/classes/question.php
<?php

class Question extends CreativeWork implements JsonSerializable {

	public static $context = 'http://schema.org';
	public static $type = 'Question';
	
	/**
	 * The answer that has been accepted as best, typically on a Question/Answer site. Sites vary in their selection mechanisms, e.g. drawing on community opinion and/or the view of the Question author.
	 * see : https://schema.org/acceptedAnswer
	 * @var \Answer|\Answer[]
	 */
	public var $accepted_answer;
	
	/**
	 * The number of answers this question has received.
	 * see : https://schema.org/answerCount
	 * @var integer|integer[]
	 */
	public var $answer_count;
	
	/**
	 * The number of downvotes this question, answer or comment has received from the community.
	 * see : https://schema.org/downvoteCount
	 * @var integer|integer[]
	 */
	public var $downvote_count;
	
	/**
	 * An answer (possibly one of several, possibly incorrect) to a Question, e.g. on a Question/Answer site.
	 * see : https://schema.org/suggestedAnswer
	 * @var \Answer|\Answer[]
	 */
	public var $suggested_answer;
	
	/**
	 * The number of upvotes this question, answer or comment has received from the community.
	 * see : https://schema.org/upvoteCount
	 * @var integer|integer[]
	 */
	public var $upvote_count;
	
	public function jsonSerialize() {
		$out = array(
			'@context' => 'http://schema.org',
			'@type' => 'Question'
		);
		
		$serialized = so_json_serialize( $this->accepted_answer );
		if ( ! empty( $serialized ) ) {
			$out['acceptedAnswer'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->answer_count );
		if ( ! empty( $serialized ) ) {
			$out['answerCount'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->downvote_count );
		if ( ! empty( $serialized ) ) {
			$out['downvoteCount'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->suggested_answer );
		if ( ! empty( $serialized ) ) {
			$out['suggestedAnswer'] = $serialized;
		}
		
		$serialized = so_json_serialize( $this->upvote_count );
		if ( ! empty( $serialized ) ) {
			$out['upvoteCount'] = $serialized;
		}
		
		return $out;
	}
}
